subsys/bluetooth/host/gatt.c

```diff
@@ -2852,12 +2855,18 @@ struct bt_gatt_attr *bt_gatt_find_by_uuid(const struct bt_gatt_attr *attr,
 					  const struct bt_uuid *uuid)
 {
 	struct bt_gatt_attr *found = NULL;
-	uint16_t start_handle = bt_gatt_attr_value_handle(attr);
-	uint16_t end_handle = start_handle && attr_count ?
-			      start_handle + attr_count : 0xffff;
+	uint16_t start_handle = bt_gatt_attr_get_handle(attr);
+	uint16_t end_handle = start_handle && attr_count
+				      ? MIN(start_handle + attr_count, BT_ATT_LAST_ATTRIBUTE_HANDLE)
+				      : BT_ATT_LAST_ATTRIBUTE_HANDLE;
+
+	if (attr != NULL && start_handle == 0U &&
+	    bt_uuid_cmp(attr->uuid, BT_UUID_GATT_PRIMARY) != 0) {
```

Comment:
This makes an assumption that the first handle is always a primary service - Not 100% if that is a valid assumption